bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/handler/HDPowerViewRepeaterHandler.java

```diff
@@ -163,6 +143,57 @@ public void handleCommand(ChannelUID channelUID, Command command) {
         }
     }
 
+    private void handleColorCommand(Command command, HDPowerViewWebTargets webTargets) throws HubException {
+        if (command instanceof HSBType hsbCommand) {
+            var color = new Color(hsbCommand.getBrightness().intValue(), ColorUtil.hsbTosRgb(hsbCommand));
+            RepeaterData repeaterData = webTargets.setRepeaterColor(repeaterId, color);
+            scheduler.submit(() -> updatePropertyAndStates(repeaterData));
+            return;
+        }
+        Color currentColor = webTargets.getRepeater(repeaterId).color;
+        if (currentColor == null) {
+            return;
+        }
+        if (command instanceof PercentType brightnessCommand) {
+            Color color = applyBrightnessToColor(currentColor, brightnessCommand.intValue());
+            RepeaterData repeaterData = webTargets.setRepeaterColor(repeaterId, color);
+            scheduler.submit(() -> updatePropertyAndStates(repeaterData));
+            return;
+        }
+        if (command instanceof IncreaseDecreaseType increaseDecreaseCommand) {
+            int brightness = switch (increaseDecreaseCommand) {
+                case INCREASE -> currentColor.brightness + BRIGHTNESS_STEP_PERCENT;
+                case DECREASE -> currentColor.brightness - BRIGHTNESS_STEP_PERCENT;
+            };
+            brightness = brightness < 0 ? 0 : brightness > 100 ? 100 : brightness;
+            Color color = applyBrightnessToColor(currentColor, brightness);
+            RepeaterData repeaterData = webTargets.setRepeaterColor(repeaterId, color);
+            scheduler.submit(() -> updatePropertyAndStates(repeaterData));
```

Comment:
You have no return here while you have one at line161.
Rather than a return at line 161, you could use "else if" at lines 163 and 173.

Reply:
You are right, that looked suspicious. I have refactored it.